pkg/network/setup/podinterface_test.go

```diff
@@ -118,27 +113,28 @@ var _ = Describe("Pod Network", func() {
 			})
 		})
 		Context("Macvtap plug", func() {
-			var podnic *podNIC
+			const primaryPodIfaceName = "eth0"
+
+			var (
+				domain        *api.Domain
+				specGenerator *MacvtapLibvirtSpecGenerator
+			)
 
 			BeforeEach(func() {
+				domain = NewDomainWithMacvtapInterface("default")
+				api.NewDefaulter(runtime.GOARCH).SetObjectDefaults_Domain(domain)
 				vmi := newVMIMacvtapInterface("testnamespace", "testVmName", "default")
-				var err error
-				podnic, err = newPhase1PodNIC(vmi, &vmi.Spec.Networks[0], mockNetwork, cacheFactory, &pid)
```

Comment:
Don't we loose coverage by not calling `newPhase1PodNIC` for macvlan, slirp bindings?

Reply:
I guess we might, but, that should be unit tested in the podnic unit tests, not on the spec generators unit tests.

Reply:
> another thing I realize; none of the tests using that method - newPhase1PodNIC - are checking the type of the created spec generator is the correct one. IMO, while it might mean we can have better coverage, we're not actually checking anything about it.

Makes sense, thanks.